npcheck-review/npcheck-dec6-2021.md

```diff
@@ -0,0 +1,114 @@
+(71): The module "openapi-editor" is not tested by community CITGM runs.
+(72): The module "openapi-editor" has "192" dependencies (including sub-dependencies) which is more than the default "20".
+(73): The latest release of "nyc" was over 1 year ago
+(74): The module "nyc" seems to have no available TypeScript typings.
+(75): The module "nyc" is not tested by community CITGM runs.
+(76): The module "nyc" has "140" dependencies (including sub-dependencies) which is more than the default "20".
+
+problems: 76 (errors: 0 - warnings: 76)
+```
+
+## Notes
+* We have added a number of new checks, which has resulted in a larger number of warnings. We plan
+  to work these down.
+* With respect to (7) we are looking at web frameworks but the concensus is still the current recommendations
+* With respect to (10),(18), (36), (70), (73) we don't see a concern with those not having done a recent release.
+* With respect to (1), after review it seems that being a data delivery package no testing is needed
+* With respect to the 10 warmings about "no support", those indicate that the package does not include
+  an engines field or package-support.json file. We have no reason to believe any of those
+  flagged don't support the current LTS Node.js versions.
+* With respect to (14), (42), (30), (48) we don't believe they would be accepted into CITGM. We plan
+  to supress these warnings.
+* With respect to (4), cldr-localenames-full has no tests which makes sense for what it is and therefore
+  does not belong in CIGGM.
+* With respect to the 26 other CITGM warnings we plan to review and see which packages make sense to be
+  part of CITGM in the community.
+* With respect to the 16 warnings (which is 16/27 modules) about modules with more than 20 dependencies, we are still working out how we should review/check dependencies. None of the warnings reported make us think we should re-look at the packages reported.
+* With respect to (3) as a data delivery module, typings do not make sense.
+* With respect to (20), (25) as plugins/backends for i18next typings do not make sense.
+* With respect to (54) as a generator tool, typings do not make sense.
+* With respect to (62) as express middleware typings do not make sense.
+* With respect to (66) as an editor typings do not make sense.
+* With respect to (74) as a test module typings do not make sense.
+* With respect to (13) as a passport plug-in strategy typings are not as important as they might be for some modules. 
+* With respect to (40) we think this is most likely an issue with npcheck as we believe JEST has tests
+* With respect to (53) openapi-generator-cli has a "test" script in package.json so this seems to be an error. 
+* With respect to (68) npcheck had some bugs in how vulns were reported.  minimist has 1 medium vulnerability which is older than 4 months. It appears the CVE has been fixed in 1.2.3 or later. It's due to swagger-editor@4.0.2 pulling in an older version in its dependencies.
```

Comment:
```suggestion
the CVE has been fixed in 1.2.3 or later. It's due to swagger-editor@4.0.2 pulling in an older version in its dependencies. Our investigation shows that its due to the use of the opencollective library and then react-ace which
uses the older version of minimist.  The opencollective library says it will not get updates. This seems like a minimal risk.
``` 
```